### PR TITLE
[fix] Fixed error in ReStructuredText Check #150

### LINKED ISSUE #150: [qa] ReStructuredCheck failing in netjsonconfig, different approach is needed
The check is failing with:
```
There are some Errors in your RST file syntax 

- <string>:1: (WARNING/2) "raw" directive disabled.
```

However, I don't think the path we've taken is good.

The repositories which have sphinx documentation already have checks to ensure that documentation compiles and renders successfully, these checks simply generate the sphinx docs and they work reliably.

Now we're mixing things.
I asked to add this check because we need to check that the README files are compatible with pypi.
If we start allowing sphinx syntax in the README, we'll hit again the same issue that some of the sphinx syntax won't work on pypi.

I think the right solution is to separate the two things.

Sphinx documentation checks must be done as we already do: https://github.com/openwisp/netjsonconfig/blob/master/run-qa-checks#L8-L10
This new check should only focus on the README compatibility with pypi, removing the changes we introduced recently to make it work on openwisp-radius.
I would also rename the check to make this more explicit: we can call it `checkreadme` to ensure things don't get confused.

@KapilBansal320 what do you think?

### PR DESCRIPTION
**Pull request by devkapilbansal**
Fixes #150

### CODE REVIEW

**Comment by KapilBansal320** [comment]
@nemesisdesign please review it.

Here I removed recursive checking on the subdirectories like docs as said. Also, **ignore_sphinx** is removed here. Although I don't rename **checkrst** to **checkreadme** as I was checking other rst files in root directory too. Like CHANGES.rst

File: openwisp_utils/admin.py [hunk 1]
```diff
@@ -147,7 +144,7 @@ def receive_url(self, obj):
         )
         baseurl = self.receive_url_baseurl
         if not baseurl:
-            baseurl = '{0}://{1}'.format(self.request.scheme, self.request.get_host(),)
```
Comment (nemesisdesign) [inline openwisp_utils/admin.py:150]: there's many changes like this one which are not related to the goal of the PR, this makes history hard to follow.
Can you please separate these changes into a separate commit?

**Review by nemesisdesign — CHANGES_REQUESTED** [review]
Thanks @KapilBansal320, looks good, but there's an issue, see my comment below and let me know if you can do that change.

File: openwisp_utils/admin.py (hunk 1 above)
Comment (KapilBansal320) [reply openwisp_utils/admin.py:150]: @nemesisdesign I break down my commit in two, one for the changes and second for the formatting. Is it correct?

Comment (devkapilbansal) [reply openwisp_utils/admin.py:150]: > @nemesisdesign please review it.
> 
> Here I removed recursive checking on the subdirectories like docs as said. Also, **ignore_sphinx** is removed here. Although I don't rename **checkrst** to **checkreadme** as I was checking other rst files in root directory too. Like CHANGES.rst

@nemesisdesign I am changing the function to check only for readme instead because when I ran it on openwisp2-docs, it fails for index.rst file (as toctree is a sphinx directive and docutils doesn't recognize it).

**Comment by devkapilbansal** [comment]
> @nemesisdesign please review it.
> 
> Here I removed recursive checking on the subdirectories like docs as said. Also, **ignore_sphinx** is removed here. Although I don't rename **checkrst** to **checkreadme** as I was checking other rst files in root directory too. Like CHANGES.rst

@nemesisdesign I am changing the function to check only for readme instead because when I ran it on openwisp2-docs, it fails for index.rst file (as toctree is a sphinx directive and docutils doesn't recognize it).

**Review by nemesisdesign — APPROVED** [review]
Good work @devkapilbansal, thanks :+1:
